File: Electronics_Shop_17.Services/Database/CustomerCoupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class CustomerCoupon : IHelperId
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public int CouponId { get; set; }
        public int UsageCount { get; set; }

        //public virtual Customer Customer { get; set; }
        //public virtual Coupon Coupon { get; set; }
    }
}

File: angular17_web/src/app/models/customer-coupon/customer.coupon.ts
export class CustomerCoupon {
  id!: number;
  customerId!: number;
  couponId!: number;
  usageCount!: number;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.customerId = data.customerId;
    this.couponId = data.couponId;
    this.usageCount = data.usageCount;
  }
}
